src/Sylius/Behat/Context/Api/Admin/RemovingProductContext.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Behat\Context\Api\Admin;
+
+use Behat\Behat\Context\Context;
+use Sylius\Behat\Client\ApiClientInterface;
+use Sylius\Behat\Client\ResponseCheckerInterface;
+use Sylius\Behat\Context\Api\Resources;
+use Sylius\Behat\Service\SharedStorageInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+use Webmozart\Assert\Assert;
+
+final class RemovingProductContext implements Context
+{
+    public function __construct(
+        private ApiClientInterface $client,
+        private SharedStorageInterface $sharedStorage,
+        private ResponseCheckerInterface $responseChecker,
+    ) {
+    }
+
+    /**
+     * @When I (try to) delete the :product product
+     */
+    public function iDeleteProduct(ProductInterface $product): void
+    {
+        $this->client->delete(Resources::PRODUCTS, $product->getCode());
+    }
+
+    /**
+     * @Then the product should still exist
```

Comment:
`the product` should be automatically handled by the `SharedStorageContext` when used as parameter here 🖖 and the 46 line would not be needed